daliuge-engine/dlg/apps/pyfunc.py

```diff
@@ -498,9 +502,17 @@ def initialize_with_func_code(self):
         """
         This function takes over if code is passed in through an argument.
         """
```

Comment:
It might be helpful having more in the docstring here, as it's a bit unclear to me what combinations we are expecting:

1. The `self.func_code` is an instance of bytes
2. The `self.func_code` is a not an instance of bytes (e.g. it is a string that is not serialized) and:
    a. Can be run using `pyext.RuntimeModule.fromString`, or
    b. Can _not_ be run using `pyext`, and so requires conversion to base64 (for backwards compatibility).   

Is this a correct summary? I think what is unclear to me is what would cause the `import_using_code` method call to fail if `serialized=False` and `self.func_code` is _not_  an instance of `bytes`.